packages/cli/src/handlers/dbt/getWarehouseClient.ts

```diff
@@ -0,0 +1,263 @@
+            };
+            break;
+        case SupportedDbtAdapter.TRINO:
+            credentials = {
+                type: WarehouseTypes.TRINO,
+                host: '',
+                user: '',
+                password: '',
+                port: 5432,
+                dbname: '',
+                schema: '',
+                http_scheme: '',
+            };
+            break;
+        default:
+            assertUnreachable(
+                dbtAdaptorType,
+                `Unsupported dbt adaptor type ${dbtAdaptorType}`,
+            );
+    }
+    return credentials;
+}
+
+type GetWarehouseClientOptions = {
+    isDbtCloudCLI: boolean;
+    profilesDir: string;
+    profile: string;
+    target?: string;
+    startOfWeek?: number;
+};
+
+type GetWarehouseClientReturn = {
+    warehouseClient: ReturnType<typeof warehouseClientFromCredentials>;
+    credentials: CreateWarehouseCredentials;
+};
+
+export default async function getWarehouseClient(
+    options: GetWarehouseClientOptions,
+): Promise<GetWarehouseClientReturn> {
+    let warehouseClient;
+    let credentials;
+    if (options.isDbtCloudCLI) {
+        const dbtAdaptorType = await getDbtCloudConnectionType();
+        GlobalState.debug(`> Using ${dbtAdaptorType} client mock`);
+        credentials = getMockCredentials(dbtAdaptorType);
+        warehouseClient = warehouseClientFromCredentials({
+            ...credentials,
+            startOfWeek: isWeekDay(options.startOfWeek)
+                ? options.startOfWeek
+                : undefined,
+        });
+        const config = await getConfig();
+        // Overwrite methods that need to connect to the warehouse
+        warehouseClient.getCatalog = async (refs) =>
+            refs.reduce<Promise<WarehouseCatalog>>(async (accPromise, ref) => {
+                const acc = await accPromise; // Wait for the previous step's result
+                if (!config.context?.project) {
+                    throw new AuthorizationError(
+                        `No active Lightdash project.`,
+                    );
+                }
+                try {
+                    GlobalState.debug(
+                        `> Warehouse schema information is not available in dbt Cloud CLI. The schema ${ref.database}.${ref.schema}.${ref.table} will be fetched from the active project.`,
+                    );
+                    const fields = await getTableSchema({
+                        projectUuid: config.context.project,
+                        tableName: ref.table,
+                        schemaName: ref.schema,
+                    });
+                    acc[ref.database] = {
+                        [ref.schema]: {
+                            [ref.table]: fields,
+                        },
+                    };
+                } catch (e) {
+                    GlobalState.debug(
+                        `Failed to get schema for ${ref.database}.${ref.schema}.${ref.table}.`,
```

Comment:
I got a lot of 404s error here, should we investigate ? 

![image](https://github.com/user-attachments/assets/400a4a08-4cc2-4d9e-b0e3-ab3d3dd4c0a2)


it didn't stop me from creatint the project though.

Reply:
It is using the newly created project which doesn't have the complete warehouse credentials. So it can't find any of the schemas on the first compile.